### ISSUE TITLE
Feature Request: Xbox gamepad / xinput support

### ISSUE THREAD

**Issue by Gagert** [dethrace-labs/dethrace#343]
It would be neat to be able to enjoy Carma 1 + Splat Pack with proper controller support, making it possible to map everything to an Xbox gamepad / xinput device. Perfect for couch gaming in front of the TV.

I suppose SDL2 has support for xinput, or at least the various Xbox gamepads. Even a simple text file for button mapping on an xinput device would be more than good enough if you ask me.

**Comment by jeff-1amstudios** [comment]
Yep, this is on my mental list. I don’t have a gamepad currently but was planning to get one to implement this unless someone beats me to it!

carma has support for joysticks, so it should be a simple ish matter of hooking up gamepad input to the carma joystick stubs

**Comment by Gagert** [comment]
I haven't personally played Carmageddon 1 with a joystick before, but I wonder how the game originally handled a joystick? Is it only the various keyboard button commands mapped to a joystick, or was it more fancy like 360 degree movement comparable to the analogue stick on an Xbox gamepad instead of just left, right, reverse and forward?

This might sound like a dumb question but way back then in the late 90s I barely touched any kind of relevant joystics or gamepads for PCs (keyboard & mouse only back then).

If Carmageddon 1 supported 360 degree movement in its code, this could be very beneficial for modern Xbox gamepads.

**Comment by JYouren** [comment]
Way back when I purchased an original carmageddon copy in a box it actually came with a free pc controller! From memory it was very similar to sega mega drive/genesis type controllers. It was fairly cheap being free, and I don't remember using it much, preferring the keyboard.

**Comment by NebularNerd** [comment]
I've played Carmageddon with many controllers over the years. It fully supports proper analogue steering and acceleration. One of my happiest times playing this was with a steering wheel and pedals years ago. Used to be great fun for hurtling around.

You can map most functions to the buttons as memory serves, if you had enough buttons you rarely needed the keyboard during play.

**Comment by fluddsskark** [comment]
is it implemented now?

**Comment by Serasul1984** [comment]
> Yep, this is on my mental list. I don’t have a gamepad currently but was planning to get one to implement this unless someone beats me to it!
> 
> Original game has support for joysticks, so it should be a simple ish matter of hooking up gamepad input to the original joystick stubs

i played C1 C2 and Max damage on steam recently  all work with the steam controller tool  (joy 2 key)

but when you test C1 and C2 outside of steam as a standalone version, only max damage works fine with a joy 2 key tool the older ones get weird bugs or dont react.  
ans playing C1 only with keyboard is horrible because of the bad steering and because there are to much keys.

full controller support where every button can be set, is a must

**Comment by eisnerguy1** [comment]
Would love this feature to be added 👍🏻

**Comment by mikedebian** [comment]
Adding my support for this. Would be great to use my 8bitdo Ultimate gamepads for this!

**Comment by petepestilence-stack** [comment]
i support this also, tried to get steam to take over, but steam overides wouldnt work for some reason

**Comment by petepestilence-stack** [comment]
^my steam issue seems to be a wayland one, so native gamepad support would be extremely appreciated

**Comment by joshewah777** [comment]
Also throwing in my support of this request. I'd be happy to use WASD as well, but the S key seems to not be re-mappable?

**Comment by fluddsskark** [comment]
> Also throwing in my support of this request. I'd be happy to use WASD as well, but the S key seems to not be re-mappable?

the s key is hardcoded in the game code to act as a sound and music toggle key that turns on and off
